File: ProjecIntegration.Api/ProjecIntegration.Api/Models/Entity/Catalogue.cs
using ProjecIntegration.Api.Models.BaseEntity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

 namespace ProjecIntegration
{
    public class Catalogue:BaseEntity
    {
       
        public string Name { get; set; }
        public string Description { get; set; }
        public IList<Representation> Representations { get; set; }
        public Complexe Complexe { get; set; }
        public int ComplexeId { get; set; }
        public DateTime DebutCatalogue { get; set; }

        public DateTime FinCatalogue { get; set; }
    }
}

File: ProjecIntegration.Api/projectintegrationFront/src/app/class/catalogue.ts
import { Complexe } from "./complexe";
import { Representation } from "./representation";

export class Catalogue {Name: string;
  Description: string;
  Representations: Representation[];
  Complexe: Complexe;
  ComplexeId: number;
  DebutCatalogue: Date;
  FinCatalogue: Date;

  constructor(
    Name: string,
    Description: string,
    Representations: Representation[],
    Complexe: Complexe,
    ComplexeId: number,
    DebutCatalogue: Date,
    FinCatalogue: Date
  ) {
    this.Name = Name;
    this.Description = Description;
    this.Representations = Representations;
    this.Complexe = Complexe;
    this.ComplexeId = ComplexeId;
    this.DebutCatalogue = DebutCatalogue;
    this.FinCatalogue = FinCatalogue;
  }
}
